In [32]:
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_log_error
from sklearn.metrics import mean_squared_error
import pandas as pd

In [34]:
df = pd.read_csv("training_dataset_dp2.csv")

In [35]:
X = df[['Num_escuelas',
 'Num_hospitales',
 'Num_iglesias',
 'Num_zonas_tipicas',
  "Hora_llegada_1",                                          
  "Hora_llegada_2",
  "Hora_llegada_3",                                          
  "Hora_llegada_4",
  "Hora_llegada_5",                                          
  "Hora_llegada_6",
  "Hora_llegada_7",                                          
  "Hora_llegada_8",
  "Hora_llegada_9",                                          
  "Hora_llegada_10",
  "Hora_llegada_11",                                          
  "Hora_llegada_12",
  "Hora_llegada_13",                                          
  "Hora_llegada_14",
  "Hora_llegada_15",                                          
  "Hora_llegada_16",                                         
  "Hora_llegada_17",                                          
  "Hora_llegada_18",
  "Hora_llegada_19",                                          
  "Hora_llegada_20",
  "Hora_llegada_21",                                          
  "Hora_llegada_22",
  "Hora_llegada_23", ]]

y = df["Num_viajes"]

In [36]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.15, random_state=42)

In [37]:
linear_model = LinearRegression()
linear_model.fit(X, y)

LinearRegression()

In [38]:
from sklearn.metrics import mean_absolute_error as mae
from sklearn.metrics import mean_squared_log_error as msle
import numpy as np 

yfit_linear = pd.DataFrame(linear_model.predict(X_train), index = X_train.index, columns = ["Num_viajes"]).clip(0.)

ypred_linear = pd.DataFrame(linear_model.predict(X_test), index = X_test.index, columns = ["Num_viajes"]).clip(0.)


print('Linear RMSLE :', np.sqrt(msle(y_test, ypred_linear)))
print('Linear MAE :', mae(y_test, ypred_linear))


Linear RMSLE : 1.4360590284591315
Linear MAE : 862.5127815297449


# Evaluación del modelo

Debido a que seguiremos iterando y evolucionando el modelo, las siguientes métricas servirán como benchmarck para saber si la siguiente iteración representa una mejora

Linear RMSLE : 1.4360590284591315

Linear MAE : 862.5127815297449
